### PR TITLE
Integration Testing Framework

### PR DESCRIPTION
**Pull request by Doy-lee**
This sets up some hooks that intercept the stdin/stdout so we can IPC to the daemon/wallet programatically from another program. Then we can remotely issue commands and verify the output for testing purposes, simulating input into the program.

This is only enabled when compiled with LOKI_ENABLE_INTEGRATION_TEST_HOOKS.

It works by reading and writing to a shared memory region, where then the controller issues commands by writing to the stdin memory region and parsing the stdout from its memory region respectively.

### CODE REVIEW

**Comment by Doy-lee** [comment]
Ready for review @msgmaxim

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -3295,7 +3328,7 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
     fail_msg_writer() << tr("Can't specify more than one of --testnet and --stagenet");
     return false;
   }
-  const network_type nettype = testnet ? TESTNET : stagenet ? STAGENET : MAINNET;
```
Comment (msgmaxim) [inline src/simplewallet/simplewallet.cpp]: I liked that const.

File: src/daemon/rpc_command_executor.cpp [hunk 2]
```diff
@@ -2174,50 +2202,101 @@ static void print_service_node_list_state(cryptonote::network_type nettype, int
...
+        result.append("\n");
       }
     }
 
-    tools::msg_writer() << "";
+    if (i < list.size())
+      result.append("\n");
   }
+
+  return std::move(result);
```
[the remark this answers is not in the document]
Comment (Doy-lee) [reply src/daemon/rpc_command_executor.cpp]: Removed std::move

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
Comment (Doy-lee) [reply src/simplewallet/simplewallet.cpp]: Added back in

**Pull request merged**